### ISSUE TITLE
Language Is Not All You Need: Aligning Perception with Language Models

### ISSUE THREAD

**Issue by reonaledo** [iRAIL-SNU/irail-paper-review#27]
### Motivation

- Microsoft의 가장 최근 발표된 Multimodal Large Language Model. KOSMOS-1
- 싱글모달 모델에 비해 멀티모달 모델이 갖는 특징들을 파악할 수 있음.

### Abstract

- fine-tuning 없이 다양한 멀티모달, 싱글모달 테스크에서 높은 성능 보임
  - Language understanding, generation, OCR-free NLP(텍스트가 있는 이미지를 바로 사용)
  - Perception-language task(multimodal dialogue, image captioning, VQA 등) 
  - Vision tasks(image recognition with descriptions 등) 
  
- Language only <-> Vision-language 간 knowledge transfer 효과를 보임
- Nonverbal reasoning task(Raven IQ test) 제안, 제안모델은 이에 어느정도의 성능을 보임(낮은 성능이긴함)

![image](https://user-images.githubusercontent.com/58892327/226499867-5ffa8d42-0154-4c77-97c0-91a13f57fe30.png)

### Introduction: From LLMs to MLLMs
- GPT-3와 같이 LLM은 다양한 자연어 태스크에 general-purpose interface로서 성공적으로 작동한다.
- 하지만 Multimodal data에 대해 여전히 LLM은 고전하고 있으며, 모델이 현실에 기반한 지식을 획득하여 AGI가 되기 위해서는 Multimodal perception 능력은 필수적이다. 최근 Frozen, MetaLM, BEiT, Flamingo, CM3, BLIP-2 의 연구에서와 같이 멀티모달 인풋은 인공지능이 할 수 있는 태스크의 범위를 크게 확장시켰다.
- 본 연구에서는 MLLM인 KOSMOS-1을 제안하며 이는 이미지를 인지하고 instruction을 따를 수 있으며(zero-shot) in-context learning이 가능하다(few-shot). 
- 연구의 목표는 인지(perception)를 LLM과 align하는 것이며 그럼으로써 모델은 보고 말할 수 있다. 
- Transformer decoder를 general-purpose interface로 사용하고, 각 모달리티의 인지모듈을 그것에 붙여 사용하는 METALM의 철학을 따른다.
- 학습 데이터는 텍스트, 이미지-캡션 pair, 이미지 텍스트가 혼합(interleaved  images and texts)를 이용
- 검증 태스크 대분류: Language task, Cross-modal transfer, Nonverbal reasoning, Perception-language tasks, Vision task

- From LLMs to MLLMs:
  - MLLM은 artificial general intelligence(AGI)로 향하는 길이다
  - 멀티모달 인지의 중요성
    1) LLM이 텍스트 디스크립션 너머의 commonsense knowledge를 얻을 수 있게 한다.
    2) Robotics, document intelligence와 같은 새로운 태스크들을 가능하게 한다.
    3) 가장 사용성이 좋아 널리 사용되는 GUI에 기반한 API를 만들 수 있다. (e.g. 컴퓨터 화면이나 영수증을 MLLM이 바로 읽음)
- Language models as general-purpose interfaces:
  - METALM의 철학을 따라 Language model을 universal task layer로 취급한다. Open-ended output space를 갖는 언어모델의 특징으로 인해 텍스트를 이용하여 다양한 태스크를 수행할 수 있다.
  - 자연어 instruction과 action sequence(프로그램언어 등)를 다루기에 language model이 적합하다
  - LM은 복잡한 태스크를 수행하기 위한 추론능력을 제공하는 Basic reasoner로써 기능한다.
- New capabilities of MLLMs:
  - Zero/few-shot learning을 통해 멀티모달 태스크를 수행할 수 있다.
  - 비언어 추론문제(이미지 추론문제)를 해결할 가능성을 보인다
  - Multimodal dialogue와 같은 멀티모달 multi-turn interaction을 가능하게 한다.

#### KOSMOS-1: A Multimodal Large Language Model
- 모델 구조는 단순함. 트랜스포머 디코더를 백본으로 하고 이미지는 vision encoder를 통해 임베딩되어 백본모델에 워드임베딩과 같이 인풋으로 들어감.
- Input Representation(아래 figure)
![image](https://user-images.githubusercontent.com/58892327/226502923-ea995c20-e611-4b7d-b8ee-2395d8521ffb.png)
- 기본 Transformer와 차이점
  1. MAGNETO(#24 )의 요소 적용: 
    - LayerNorm을 매 linear layer 이전에 추가. 즉, 하나의 transformer 레이어에서 4번의 LN. 
    - 모델의 스케일을 키울 때 안정적인 parameter initialization 제안방법론 적용
  2. xPOS(#28 ):
    - 긴 텍스트에 적합하며, 트레인-인퍼런스 간 텍스트 길이 차이에 강건한 포지션 인코딩 사용.
- 학습 목적은 **텍스트 토큰에 대한** next-token prediction.

#### Model Training
- 데이터: 웹스케일 데이터. 디테일은 생략
  -  Text copra
  - Image-caption pairs
  - interleaved data of images and texts : 웹페이지 크롤링. 2B개 웹페이지 중 적절한 71M개 웹페이지 골라 사용
  
![image](https://user-images.githubusercontent.com/58892327/226510721-293fe8c7-0212-43c6-a3a8-bbdec3d677ea.png)
- Training Setup
  - Transformer decoder: 
    - 파라메터 총 1.3B개
    - SentencePiece tokenizer
    - RoBERTa's full-sentence format: 인풋 시퀀스를 max length에 맞춰 이어붙여 넣음
  - Vision encoder는 Pretrained CLIP ViT-L/14. 1024 feature dim. 224*224 resize. Freeze하고 마지막 레이어만학습
  - 총 파라메터 1.6B.

- Language-Only Instruction Tuning
  - Instruction Tuning이란 언어모델이 instruction을 따라 주어진 문제를 zeroshot으로 풀기 위해 Instruction과 Input(문제), Output(정답)으로 이루어진 데이터셋을 이용해 언어모델에 학습시키는 것.
  - 오직 Output에 해당하는 토큰에 대해서만 loss를 계산함
  - KOSMOS-1은 Instruction tuning 데이터셋으로 Unnatrual Instructions와 FLANv2를 사용
  - Unnatrual Instruction은 언어모델을 통해 인스트럭션, 인풋, 아웃풋을 모두 생성한 합성데이터셋
![image](https://user-images.githubusercontent.com/58892327/226512714-d67d1e65-f8b1-4c42-85cb-e78769fb7914.png)
  - FLANv2는 기존의 다양한 NLP 테스크를 이용해 Instruction, input, output으로 구성한 데이터셋.
 

### Experiment & Result

검증 태스크: 
- Language tasks
  - Language understanding
  - Language generation
  - OCR-free text classification
- Cross-modal transfer
  - Commonsense reasoning
- Nonverbal reasoning
  - IQ Test (raven's Progressive Matrices)
- Perception-language tasks
  - Image captioning
  - Visual question answering
  - Web page question answering
- Vision tasks
  - Zero-shot image classification
  - Zero-shot image classification with descriptions 

Baseline LLM:
- KOSMOS-1의 멀티모달 학습 효과를 보기 위하여 이와 동일한 셋업으로 LLM을 훈련시켜 몇몇 테스크에서 baseline으로 활용.

#### Perception-Language Tasks
- Image captioning: 
![image](https://user-images.githubusercontent.com/58892327/226515288-380ef826-0922-4ca6-811b-f8ad8fabcb5b.png)
![image](https://user-images.githubusercontent.com/58892327/226515325-ecf260e7-7018-49f2-b773-9021f17ca5f3.png)
![image](https://user-images.githubusercontent.com/58892327/226515350-41915aea-8dbf-4f23-b4ed-87a114b7fa8b.png)
  - Flamingo와 비교할 때 파라메터 수 훨씬 적지만 더 좋은 성능. 왜 더 잘나올까? 데이터? 모델 구조?

- Visual Question Answering
![image](https://user-images.githubusercontent.com/58892327/226515951-732a9245-25a6-4476-9549-9e35a29f9213.png)
![image](https://user-images.githubusercontent.com/58892327/226516050-8cb2bfb1-b736-42e8-b525-ea4911e3ccec.png)
  - 마찬가지로 파라메터 수 훨씬 많은 Falmingo와 비슷한 성능
 
- Web Page Question Answering
  - 웹페이지 이미지와 그 텍스트로부터 답을 찾는 태스크. 웹페이지 내의 테이블, 목록, 레이아웃 등을 파악하여 의미를 이해해야 함. 
  - 웹페이지의 텍스트를 가져와 아래와 같이 프롬프트 구성하여 LLM에 인풋. KOSMOS-1은 이 앞에 웹페이지 이미지를 추가함
“Given the context below from web page, extract the answer from the given text like this: Qusestion: Who is the publisher of this book? Answer: Penguin Books Ltd. Context: {_WebText_} Q: {_question_} A: {_answer_} ”
![image](https://user-images.githubusercontent.com/58892327/226517018-93492b63-013e-4e71-a792-785d89045bba.png)
  - LLM보다 나은 성능으로 보아 웹페이지의 시각적 구조로부터 정보를 얻는 것이 성능향상에 큰 도움이 될 수 있음을 확인
  - 하지만 아직 텍스트 없이 이미지만 넣을 때 매우 낮은 성능.

#### IQ Test: Nonverbal Reasoning
![image](https://user-images.githubusercontent.com/58892327/226517233-dc35f09f-dc3c-41cf-b58f-9aacf08941ae.png)
- Raven IQ Test는 매트릭스 형태로 배치된 도형을 보고 빈칸에 적합한 도형을 고르는 테스트.
- 텍스트 형식의 추론이 아닌 비언어적 추론이 가능한지 확인하기 위함.
- 각 도형을 이미지로 하여 텍스트 프롬프트와 함께 일렬로 늘어놓고 각 선택지 도형의 이미지와 그에 대한 Is it correct?프롬프트 바로 뒤에 Yes가 올 확률을 추출, 가장 Yes 확률이 높은 도형을 선택하는것으로 태스크를 구성.
![image](https://user-images.githubusercontent.com/58892327/226517589-57937f15-31f5-41ab-a671-4be0cce7ae6a.png)
- Random choice에 비해 나아진 성능을 보이는 정도. Nonverbal reasoning의 첫 시도이기 때문에 MMLM이 이 태스크에 도움이 된다는 것을 확인한 것만으로도 의미가 있다고 생각함. 
- language-only instruction tuning을 안 했을 때 성능이 더 좋음. --> instruction을 따르는 능력이 언어에 국한하여 튜닝되는 듯.

#### OCR-Free Language Understanding
- OCR 없이 NLP 문제(감성분석 등))가 텍스트로 적혀 있는 이미지를 그대로 인식하여 문제를 푸는 태스크
- 2가지 벤치마크 사용
  - Rendered SST-2: 기존 감성분석 태스크인 SST-2를 이미지로 랜더링 한 것. 이미지 -> 긍/부정 예측 
  - HatefulMemes(#25 ): 악의가 담긴 Meme 이미지를 구별(이진분류)하는 벤치마크. 
![image](https://user-images.githubusercontent.com/58892327/226518432-79507a02-557a-4bb9-9aa5-a89cbc14a787.png)

![image](https://user-images.githubusercontent.com/58892327/226518485-85268095-2a42-4df6-aaee-01d751d82cb9.png)
- KOSMOS-1이 가장 좋은 성능. 특히 Flamingo는 OCR을 통한 텍스트를 인풋으로 함께 넣어준 결과임.
- HatefulMemes SOTA(finetuned), SST-2 SOTA(finetuned)는 각각 85.8%, 97.5%수준이지만 MMLM은 zero-shot 이미지만으로 이 태스크 수행이 어느정도 가능하다는 것을 보여줌.

#### Multimodal Chain-of-Thought Prompting
- Chain-of-thought(#26)
  - Language Model(LM)이 아웃풋으로 정답만을 출력하는 것이 아니라 정답에 다다르는 여러 단계의 논리적 과정을 생성한 뒤 정답을 생성하도록 example prompt를 입력하는 방식을 제안. 
  - 즉 모델이 문제의 정답만 생성하는게 아니라 풀이과정까지 생성하게 함.
  - 모델 사이즈가 커질때(~100B) 일반 프롬프팅과 비교하여 추론 성능에서 매우 강력한 효과를 보임
![image](https://user-images.githubusercontent.com/58892327/226291663-1d780f4d-85b5-4112-8bdc-fdc7e69018a4.png)

- CoT에 영감을 받아 Multimodal CoT 제안. 
- 아래 Figure5와 같이 2개 단계에 걸쳐 진행됨
  1. KOSMOS-1로 이미지를 설명하는 텍스트(rationale) 생성
  2. rationale과 instruction 프롬프트를 다시 KOSMOS-1에 넣어 정답 생성. Image는 다시 넣지 않음.
![image](https://user-images.githubusercontent.com/58892327/226521847-73119b9c-1ccb-417a-84b7-b67bfb970143.png)
- 그 성능을 확인하기 위해 위의 Rendered SST-2 벤치마크에 대해Multimodal CoT 사용 전/후를 비교.
![image](https://user-images.githubusercontent.com/58892327/226522696-72c44dd1-c401-4d24-919e-8ac6d0634530.png)
- Multimodal CoT를 사용했을 때 성능 향상의 폭이 컸다. 
- CoT처럼 Multimodal CoT도 이미지를 중간 컨텐츠로 변환하는 단계를 거침으로써 이득이 되더라.

#### Zero-shot Image Classification
- Generation-based image classification: CLIP과 같이 이미지 임베딩과 "The photo of the {label}"의 프롬프트 임베딩간의 유사도를 통해 클래스를 예측하는 것이 아닌, 이미지와 프롬프트("The photo of the ")를 입력했을 때 language modeling으로 vocab 내의 단어를 생성함으로써 클래스를 예측하게 함. GIT(2022, Microsoft) 논문에서 처음 제안함
- 이미지넷 성능에서 GIT를 상회하는 성능. GIT는 KOSMOS-1과 유사한 모델 구조지만 학습방식에서 조금 차이가 있음. 다양한 Multimodal 테스크에서 2022년 8월 기준으로 SOTA를 달성했음. ImageNet에서는 Finetuning 시 88.8 top-1 acc(CoCa가 91.0으로 SOTA).
- vocab을 레이블들만으로 한정지을 때 성능향상 두드러짐.

#### Zero-shot Image Classification with Descriptions
![image](https://user-images.githubusercontent.com/58892327/226526528-309a3526-1280-4636-8233-57b2f2ea9fd2.png)
- 그림과 같이 디스크립션을 함께 넣어 줄 경우 성능이 크게 향상됨.
![image](https://user-images.githubusercontent.com/58892327/226526619-b6651f2a-1723-4d53-a67f-2c499a1cac99.png)
- 즉, KOSMOS-1은 instruction의 의도를 파악하고 그것에 텍스트로 표현된 개념과 시각적 특성(visual feature)를 연관지을 수 있음.

#### Language Tasks
- LLM과 비교했을 때 MLLM인 KOSMOS-1에게 어떤 변화가 보이는지 실험.
![image](https://user-images.githubusercontent.com/58892327/226526774-c365af6b-391a-4c00-93cb-6129448d91ea.png)
- LLM과 평균적으로 유사한 성능 보임. 성능이 상승하진 않네?

#### Cross-modal Transfer
- 모달리티 간의 지식을 전달하는것이 가능한지 실험.

1. Language to Multimodal
  - Language-only Instruction Tuning을 통한 Image captioning과 VQA 성능 변화를 관찰
![image](https://user-images.githubusercontent.com/58892327/226527251-b49d3c8f-1375-42f1-ae1f-c1503bc45840.png)
  - Language-only Instruction Tuning을 통해 Instruction을 해석하고 따르는 능력을 강화한 것이 모달리티를 아울러 확인됨.
  
2. Multimodal to Language
  - LLM에 비해 KOSMOS-1이 _**텍스트만으로**_ 구성된 시각적 추론문제(Visual commonsense reasoning)를 더 잘 해결하는지 관찰
    1. Size Reasoning: 두 물체 간의 크기를 비교하는 텍스트를 프롬프트로 넣음. 이진분류
    2. Color Reasoning: 대상의 색깔을 물어보는 텍스트를 프롬프트로 넣음. 11개 색 레이블
![image](https://user-images.githubusercontent.com/58892327/226528410-fb65a598-123e-4588-be7a-eaedd1ce02ff.png)
  - BERT: 34.8, 24.3, 26.3
  - GPT-2: 47.2, 44.1, 39.1
  - CLIP: 48.0, 26.3, 23.1
  - VisualBERT: 67.2, 26.7, 38.0

- KOSMOS-1이 LLM에 비해 높은 성능. 시각적 모달리티로부터 획득한 지식을 언어 태스크에 활용하는것이 가능하다. 
- LLM이 언어만으로 시각적 정보를 학습하는데는 한계가 있다. 

### Insight

- 본 제안모델이 모든 태스크에서 뛰어난 성능을 보이는 것은 아니지만 MLLM이 LLM에 비해 갖을 수 있는 장점을 잘 분석해놓았다. 
- 본 논문에서 사용한 모델의 구조, 학습방식, 공학적 구성요소, 검증방식을 앞으로의 연구에 참고할 수 있을 것.
- Transformer Decoder를 universal task layer로 해석하는 철학이 설득력 있다는 것을 느낌. --> GPT-3과 같은 Causal language model이 사람과 같이 다양한 NLP문제를 잘 해결하는 것을 통해 NLP 문제를 language modeling 으로 reduce하는 방식이 잘 작동한다는 의견을 들었다. 하지만 사실 사람의 사고방식 자체가 애초에 language modeling과 같은 형태이기 때문에 그것이 잘 작동했던 것은 아닐까? 하는 생각을 했다. 이런 생각은 본 논문에서 "Decoder가 Universal Task Layer다" 라는 철학과 일치한다. 사람의 생각은 language modeling 처럼 sequential하게 생성되기 때문에 decoder가 사람의 생각을 잘 모델링하는 것이 아닐까?
- 그런데 사람의 생각은 언어뿐만 아니라 세상의 모든 모달리티로부터 시작될 수 있다. 즉, 이미지를 보면서도 사람은 시퀀셜하게 생각하는것 아닐까? 그렇다면 이미지를 시퀀셜하게 모델링하는 방법을 찾는다면 Transformer Decoder가 텍스트만을 Loss로 사용하여 학습하는 것이 아니라 이미지도 Loss로 사용할 수 있지 않을까? 이미지 뿐만 아니라 다른 모달리티들도 시퀀셜하게 모델링하는 방법을 찾는다면, 하나의 Transformer Decoder가 모든 종류의 모달리티 인풋으로부터 generation loss를 사용하여 학습하여 특정 모달리티에 국한하지 않은 "생각" 그 자체를 모델링 할 수 있는 게 아닐까?

### Paper URL

https://arxiv.org/pdf/2302.14045v1.pdf
